mlx_graphs/datasets/ogb_dataset.py

```diff
@@ -0,0 +1,194 @@
+from typing import Literal, Optional, Union, get_args, overload
+
+import mlx.core as mx
+import numpy as np
+
+from mlx_graphs.data.data import GraphData
+from mlx_graphs.datasets import Dataset
+from mlx_graphs.utils import index_to_mask
+
+OGB_NODE_DATASET = Literal[
+    "ogbn-products",
+    "ogbn-proteins",
+    "ogbn-arxiv",
+    "ogbn-papers100M",
+    # "ogbn-mag", # TODO: requires heterogeneous graphs
+]
+OGB_EDGE_DATASET = Literal[
+    "ogbl-ppa",
+    "ogbl-collab",
+    "ogbl-ddi",
+    "ogbl-citation2",
+    # "ogbl-wikikg2",  # TODO: requires heterogeneous graphs
+    # "ogbl-biokg",  # TODO: requires heterogeneous graphs
+    "ogbl-vessel",
+]
+OGB_GRAPH_DATASET = Literal[
+    "ogbg-molhiv",
+    "ogbg-molpcba",
+    "ogbg-ppa",
+    "ogbg-code2",
+]
+
+ALL_DATASETS = Literal[
+    *[
+        i
+        for i in list(get_args(OGB_NODE_DATASET))
+        + list(get_args(OGB_EDGE_DATASET))
+        + list(get_args(OGB_GRAPH_DATASET))
+    ]
+]
+
+
+@overload
+def to_mx_array(x: None) -> None:
```

Comment:
what is the purpose of this overloading?

Reply:
to avoid having typing errors/warnings in the code below where it's used (lines 149-151 and similar)